### PR TITLE
UI for consolidation of duplicate patients

### PR DESCRIPTION
**Pull request by anvmn**
#881

### CODE REVIEW

**Comment by anvmn** [comment]
Peek 2023-09-21 15-26

File: server/hedley/modules/custom/hedley_patient/hedley_patient.module [hunk 1]
```diff
@@ -235,3 +254,260 @@ function hedley_patient_mother_birth_date_validation($form, &$form_state) {
...
+  $message = '';
+  foreach ($form_state['storage']['messages'] as $line) {
+    $message .= "<div>$line</div>";
+  }
+
+  $original_info = hedley_patient_generate_consolidated_patient_info($form_state['storage']['original']);
+  $duplicate_info = hedley_patient_generate_consolidated_patient_info($form_state['storage']['duplicate']);
+  $message .= '<br>';
+  $message .= "<div>Original patient: $original_info</div>";
+  $message .= "<div>Duplicate patient: $duplicate_info</div>";
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_patient/hedley_patient.module]:
```suggestion
  $message .= '<div>' . t('Original patient: @original_info', ['@original_info' => $original_info]) . '</div>';
  $message .= '<div>' . t('Duplicate patient: @duplicate_info', ['@duplicate_info' => $duplicate_info]) . '</div>';
```

File: server/hedley/modules/custom/hedley_patient/hedley_patient.module [hunk 2]
```diff
@@ -235,3 +254,260 @@ function hedley_patient_mother_birth_date_validation($form, &$form_state) {
...
+ *   name and birth date.
+ *
+ * @throws EntityMetadataWrapperException
+ */
+function hedley_patient_generate_consolidated_patient_info($nid) {
+  $node = node_load($nid);
+  $name = hedley_person_get_paitent_name($node);
+  $wrapper = entity_metadata_wrapper('node', $node);
+  $birth_date = date("Y-m-d", $wrapper->field_birth_date->value());
+  return "Name: $name, Birth date: $birth_date";
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_patient/hedley_patient.module]:
```suggestion
  return t('Name: @name, Birth date: @birth_date', ['@name' => $name, '@birth_date' => $birth_date]);
```

File: server/hedley/modules/custom/hedley_person/hedley_person.module [hunk 3]
```diff
@@ -1429,3 +1411,31 @@ function hedley_person_content_affected_by_deletion($person_id) {
...
+ */
+function hedley_person_get_paitent_name($node) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+  $first_name = trim($wrapper->field_first_name->value());
+  $second_name = trim($wrapper->field_second_name->value());
+  if (empty($first_name) && empty($second_name)) {
+    $second_name = $wrapper->label();
+  }
+
+  return empty($first_name) ? $second_name : "$second_name $first_name";
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_person/hedley_person.module:1440]: Can it be the case that the first name is filled, but the second name is empty?

Comment (anvmn) [reply server/hedley/modules/custom/hedley_person/hedley_person.module:1440]: @AronNovak 
Second name is a required field while first name is not. So, second name should not be empty.
These 2 fields were added to Person type after a while, so there are patients that got both of them empty, and the name was set as node title.

**Review by AronNovak — APPROVED** [review]

**Pull request merged**